[Cocoa] Enable reordering size computation for annexB H264 and H265 decoders

youennf opened the pull request:
#### 2ccf91f163fb0909f092aafd36f8d8cbe6d74a3f
<pre>
[Cocoa] Enable reordering size computation for annexB H264 and H265 decoders
<a href="https://rdar.apple.com/122902399">rdar://122902399</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=269462">https://bugs.webkit.org/show_bug.cgi?id=269462</a>

Reviewed by Eric Carlson.

The SPS parser is starting after the NALU header and not at the NALU directly.
We offset by 1 for H264 and by 2 for H265 as per spec.

We also need to do this parsing only for annexB and not AVC/HEVC formats.
Minor cleanup to rename _useAVC by _useHEVC in RTCVideoDecoderH265.

Covered by added tests.

* LayoutTests/http/tests/webcodecs/h264-reordering-annexB-expected.txt: Added.
* LayoutTests/http/tests/webcodecs/h264-reordering-annexB.html: Added.
* LayoutTests/http/tests/webcodecs/hevc-reordering-annexB-expected.txt: Added.
* LayoutTests/http/tests/webcodecs/hevc-reordering-annexB.html: Added.
* LayoutTests/media/media-source/content/test-bframes-hevc.mp4.annexb: Added.
* LayoutTests/media/media-source/content/test-fragmented.mp4.annexb: Added.
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
(-[RTCVideoDecoderH264 decodeData:size:timeStamp:]):
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH265.mm:
(ComputeH265ReorderSizeFromAnnexB):
(-[RTCVideoDecoderH265 init]):
(-[RTCVideoDecoderH265 decodeData:size:timeStamp:]):
(-[RTCVideoDecoderH265 setHVCCFormat:size:width:height:]):
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/nalu_rewriter.cc:

Canonical link: <a href="https://commits.webkit.org/275173@main">https://commits.webkit.org/275173@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b372bd16e0f4f12e66f28e5108efba9e8ffd2465

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41085 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20098 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43463 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43647 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37177 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23123 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17429 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34015 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41659 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17021 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35385 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14649 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/14770 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36384 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44961 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37276 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36697 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/40447 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15900 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13035 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38823 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17519 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/35664 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9224 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17571 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17163 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson reviewed (approved):
Nit: Commit message is incomplete: 

    We offset by 1 for H264 and by _2_ for H265 as per spec

youennf committed:
[Cocoa] Enable reordering size computation for annexB H264 and H265 decoders
rdar://122902399
https://bugs.webkit.org/show_bug.cgi?id=269462

Reviewed by Eric Carlson.

The SPS parser is starting after the NALU header and not at the NALU directly.
We offset by 1 for H264 and by 2 for H265 as per spec.

We also need to do this parsing only for annexB and not AVC/HEVC formats.
Minor cleanup to rename _useAVC by _useHEVC in RTCVideoDecoderH265.

Covered by added tests.

* LayoutTests/http/tests/webcodecs/h264-reordering-annexB-expected.txt: Added.
* LayoutTests/http/tests/webcodecs/h264-reordering-annexB.html: Added.
* LayoutTests/http/tests/webcodecs/hevc-reordering-annexB-expected.txt: Added.
* LayoutTests/http/tests/webcodecs/hevc-reordering-annexB.html: Added.
* LayoutTests/media/media-source/content/test-bframes-hevc.mp4.annexb: Added.
* LayoutTests/media/media-source/content/test-fragmented.mp4.annexb: Added.
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm:
(-[RTCVideoDecoderH264 decodeData:size:timeStamp:]):
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH265.mm:
(ComputeH265ReorderSizeFromAnnexB):
(-[RTCVideoDecoderH265 init]):
(-[RTCVideoDecoderH265 decodeData:size:timeStamp:]):
(-[RTCVideoDecoderH265 setHVCCFormat:size:width:height:]):
* Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/nalu_rewriter.cc:

Canonical link: https://commits.webkit.org/275173@main
